test/common/upstream/logical_dns_cluster_test.cc

```diff
@@ -115,7 +116,7 @@ TEST_P(LogicalDnsParamTest, ImmediateResolve) {
   )EOF";
 
   EXPECT_CALL(initialized_, ready());
-  EXPECT_CALL(dns_resolver_, resolve("foo.bar.com", std::get<1>(GetParam()), _))
+  EXPECT_CALL(*dns_resolver_, resolve("foo.bar.com", std::get<1>(GetParam()), _))
```

Comment:
Started looking at this today - what should the general strategy be here? I would need to expose the dns_resolver_ in a getter for both StrictDnsClusterImpl and LogicalDnsCluster; and then also expose the ares_channel_ from inside of DnsResolverImpl as well, yes? Then I could double check that the resolvers set in the ares_channel are what what's given in the config? Similar to: https://github.com/lyft/envoy/pull/1018/files#diff-aafe7d77f5c31414857274dfc091160cR243 ?